docs/runtime_image.md

```diff
@@ -0,0 +1,83 @@
+# Runtime image
+
+## Overview
+
+For a dynamic languages like PHP, Python or Ruby, the build-time and run-time environments are the same. In this case using builder as a base image for a resulting image is natural.
+
+For compiled languages like C/C++, Java or Golang, the dependencies necessary for compilation might dramatically outweigh the actual runtime artifacts. To keep runtime images slim, S2I enables multiple-step build processes, where a binary artifact such as an executable or Java WAR file is created in the first builder image, extracted, and injected into an second image that simply places the executable in the correct location for execution. To give you even more abilities to customize a resulting image, S2I is also executing `assemble-runtime` script inside of this (runtime) image. In this way you may do a final adjustments by modifying files before an image will be committed.
+
+The following scheme illustrates the build workflow:
+
+![s2i workflow](./runtime-image-flow.png "s2i workflow")
+
+
+## How it works?
+
+To make this work S2I needs to know the following:
+
+* builder image
+* mapping of the artifacts
+* runtime image
+
+And all these information you should specify during S2I invocation:
+
+    s2i build <repo> <builder-image> <my-app> --runtime-image <runtime-image> --runtime-artifact </path/to/artifact>
+
+The only new here are `--runtime-image` and `--runtime-artifact` options. The latter option is used to specify a full path to file (or directory) inside of the builder image and a corresponding directory on the runtime image where S2I will copy these files. Note that you can specify this option multiple times.
+
+For our example S2I will do the following steps:
+
+1. runs builder container and performs a build (as usual)
+1. after build finishes but before stopping runtime container, it downloads artifacts from the builder and places them to a temporary directory on the host
+1. runs container with runtime image
+1. uploads scripts and build artifacts from a temporary directory into `WORKDIR` of a runtime container
+1. runs `assemble-runtime` script in a runtime container
+1. commits container with runtime image
+1. stops build and runtime containers
+
+## Details
+
+### Format and behavior of `--runtime-artifact` option
+
+`--runtime-artifact` option have a format `<source>[:<destination>]`. Here is an example of sample values:
+
+| Value                   | Meaning |
+|-------------------------|---------|
+| `/tmp/app.war`          | `/tmp/app.war` file will be extracted from a builder image and uploaded into the `WORKDIR` of a runtime image |
+| `/tmp/app.war:.`        | the same as above |
+| `/tmp/app.war:app/dist` | `/tmp/app.war` file will be copied from a builder image into into the `app/dist` subdirectory of the `WORKDIR` of a runtime image |
+| `/opt/data`             | `/opt/data` directory will be copied from a builder image into the `WORKDIR` of a runtime image |
```

Comment:
> what about /opt/data/ (trailing slash). I assume then i don't get a "data" directory.

It will works exactly the same as without trailing slash because we're invoking `filepath.Clean()` on `source` and `destination`. I added this example to doc.